File: python/p026.py
"""
	
Project Euler - Problem 26 - Reciprocal cycles
https://projecteuler.net/problem=26

"""
import timing
from eulerutil import long_division

def main():
	limit = 1000
	answer = 0
	longest = 0

	for d in range(2, limit):
		r = long_division(1, d)
		l = len(r.split('(')[1])
		if l > longest:
			longest = l
			answer = d

	print(answer)


if __name__ == '__main__':
	main()

File: js/p026.js
"use strict";
var longDivision = require("./util/tools").longDivision;
/*
 * Reciprocal cycles
 * Problem 26
 * A unit fraction contains 1 in the numerator. The decimal representation of 
 * the unit fractions with denominators 2 to 10 are given:
 * 
 * 1/2	= 	0.5
 * 1/3	= 	0.(3)
 * 1/4	= 	0.25
 * 1/5	= 	0.2
 * 1/6	= 	0.1(6)
 * 1/7	= 	0.(142857)
 * 1/8	= 	0.125
 * 1/9	= 	0.(1)
 * 1/10	= 	0.1
 * Where 0.1(6) means 0.166666..., and has a 1-digit recurring cycle. It can be 
 * seen that 1/7 has a 6-digit recurring cycle.
 * 
 * Find the value of d < 1000 for which 1/d contains the longest recurring cycle 
 * in its decimal fraction part.
 * 
 * Answer: 983
 */

function init() {
    var max = 0;
    for (var d=1; d<1000; d++) {
        if (longDivision(1, d).length > max) {
            max = d;
        }
    }
    return max;
}

module.exports.init = init;